netkan/Dockerfile

```diff
@@ -1,55 +1,49 @@
-FROM ubuntu:22.04 as ubuntu_with_python
+FROM ubuntu:24.04 AS ubuntu_with_python
 
 # Don't prompt for time zone
 ENV DEBIAN_FRONTEND=noninteractive
 
 # Install Git and Python
 RUN apt-get update \
     && apt-get install -y --no-install-recommends \
-        git libffi-dev openssh-client \
-        python3 python-is-python3 \
-    && apt-get clean
+    git libffi-dev openssh-client \
+    python3 python-is-python3 ca-certificates curl && \
+    apt-get clean && \
+    curl -sSL https://bootstrap.pypa.io/get-pip.py -o get-pip.py && \
+    rm /usr/lib/python3.12/EXTERNALLY-MANAGED && \
+    python3 get-pip.py --no-input && rm get-pip.py
+RUN groupmod -n 'netkan' 'ubuntu' && \
+    usermod -l 'netkan' -d /home/netkan -m ubuntu
 
-FROM ubuntu_with_python as base
-RUN apt-get install -y --no-install-recommends \
-    python3-pip python3-setuptools python3-dev
-RUN useradd -ms /bin/bash netkan
+FROM ubuntu_with_python AS base
 ADD . /netkan
 WORKDIR /netkan
-RUN pip install pip --upgrade
 RUN chown -R netkan:netkan /netkan
 USER netkan
 RUN pip install --user . --no-warn-script-location
 
-FROM ubuntu_with_python as production
+FROM ubuntu_with_python AS production
 COPY --from=base /home/netkan/.local /home/netkan/.local
-RUN useradd -Ms /bin/bash netkan
 RUN chown -R netkan:netkan /home/netkan
 WORKDIR /home/netkan
 USER netkan
 ADD .gitconfig .
```

Comment:
I'm getting this file marked as executable in the container. Maybe due to differences with what is tracked in inodes across platforms?
Can we set a umask here or something?

```
-rwxr-xr-x 1 root   root    196 Sep 27  2021 .gitconfig
```

Reply:
We can do. Probably should use COPY in place of ADD. It's notable I didn't change this line.

https://www.docker.com/blog/docker-best-practices-understanding-the-differences-between-add-and-copy-instructions-in-dockerfiles/

Reply:
Seems ok with COPY. Probably some quirk with ADD

```
-rw-r--r--  1 netkan netkan  397 Sep 19 00:53 .gitconfig
```

Though I'm not sure how you're seeing root? We haven't used a root user for quite some time.

Reply:
I'm still seeing this, but if it works differently for you, I'm satisfied that it's a just a quirk of Docker on Windows; no need to pursue further. Also the `ADD` ⇒ `COPY` change looks like a good idea 👍 .